vm_insnhelper.c

```diff
@@ -4791,6 +4791,8 @@ vm_defined(rb_execution_context_t *ec, rb_control_frame_t *reg_cfp, rb_num_t op_
 
             if (me) {
                 VALUE klass = vm_search_normal_superclass(me->defined_class);
+                if (klass == (VALUE)NULL) return false;
```

Comment:
maybe the following would work as well?

```suggestion
                if (!klass) return false;
```